grails-app/services/org/pih/warehouse/inventory/CycleCountService.groovy

```diff
@@ -399,4 +399,46 @@ class CycleCountService {
             transaction.addToTransactionEntries(transactionEntry)
         }
     }
+
+    CycleCountItemDto updateCycleCountItem(CycleCountUpdateItemCommand command) {
+        command.cycleCountItem.quantityCounted = command.quantityCounted
```

Comment:
This method looks a bit weird. Could we do something like this
```
CycleCountItem cycleCountItem = CycleCountItem.get(command.id)

// Copy the standard parameters that are the same
cycleCountItem.properties = command.properties

// FIXME calculates the actual index based on the index of this cycle count item
command.cycleCountItem.countIndex = command.recount ? 1 : 0

// FIXME apply evan's requested changes
command.cycleCountItem.status = CycleCountItemStatus.COUNTING
cycleCountItem.date = new Date()

// return the cycle count item instead of the DTO.
return cycleCountItem
```

Reply:
There is no need to fetch the cycle count item, because it's bound in the command and accessible via `command.cycleCountItem`.
I could try to play around with the `command.properties`, but I just wanted to be explicit here so that you can immediately see what fields are editable.